pgmax/fg/graph.py

```diff
@@ -108,16 +109,16 @@ def __hash__(self) -> int:
 
     def add_factor(
```

Comment:
I just realized that after the refactor, we no longer need to implement `add_factor`, `add_factor_by_type` and `add_factor_group` this way. Instead, we can construct factors/factor groups outside the factor graph and have a single generic `add_factor` function which takes as input the constructed factors/factor groups. This is a somewhat big change (but hopefully a simple one as you just need to delete lots of code), and it should make everything much more modular/intuitive.